src/com/tony/server/WorkerThread.java

```diff
@@ -37,6 +39,21 @@ public void run() {
         }
     }
 
+    private void logRequest(String requestMessage) {
+        logger.log("Received Request:\r\n");
+        logger.log(requestMessage);
+    }
+
+    public void logResponse(Response response, byte[] responseMessage) {
+        if(response.getClass().equals(ImageContentResponse.class)){
+            logger.log(ResponseBuilder.buildResponse(
+                    response.getStatus(), response.getHeaders(), "Image body not shown in log"));
+        } else {
+            logger.log("Sending Response:\r\n");
+            logger.log(new String(responseMessage));
+        }
+    }
```

Comment:
I've been thinking about this but the way I implemented this, I would have to handle the logic either here, or in the logger like before, or I would have to let my Responses hold a logger. Although I agree that it might get messy, I think this is the best option for now!